packages/front-end/components/Features/VariationsInput.tsx

```diff
@@ -156,196 +176,47 @@ export default function VariationsInput({
-                                    ? 0
-                                    : percentToDecimal(e.target.value)
-                                );
-                                if (e.target.value === "") {
-                                  // I hate this, but not is also the easiest
-                                  setTimeout(() => {
-                                    e.target.focus();
-                                    e.target.select();
-                                  }, 100);
-                                }
-                              }}
-                              type="number"
-                              min={0}
-                              max={100}
-                              step="any"
-                              className={styles.percentInput}
-                            />
-                            <span>%</span>
-                          </div>
-                        </div>
-                      ) : (
-                        <div className="col d-flex flex-row">
-                          {decimalToPercent(weights[i])}%
-                        </div>
-                      )}
-                      {setVariations && (
-                        <div className="col-auto">
-                          <MoreMenu>
-                            {i > 0 && (
-                              <button
-                                className="dropdown-item"
-                                onClick={(e) => {
-                                  e.preventDefault();
-
-                                  const newValues = [...variations];
-                                  [newValues[i], newValues[i - 1]] = [
-                                    newValues[i - 1],
-                                    newValues[i],
-                                  ];
-
-                                  setVariations(newValues);
-                                }}
-                              >
-                                move up
-                              </button>
-                            )}
-                            {i < variations.length - 1 && (
-                              <button
-                                className="dropdown-item"
-                                onClick={(e) => {
-                                  e.preventDefault();
-
-                                  const newValues = [...variations];
-                                  [newValues[i], newValues[i + 1]] = [
-                                    newValues[i + 1],
-                                    newValues[i],
-                                  ];
+            <DndContext
+              sensors={sensors}
+              collisionDetection={closestCenter}
+              onDragEnd={({ active, over }) => {
+                if (active.id !== over.id) {
+                  const oldIndex = getVariationIndex(active.id);
+                  const newIndex = getVariationIndex(over.id);
```

Comment:
Nitpick:

We can simplify this a bit since `getVariationIndex` is looping through the array each time we call it.

Since our IDs are based on the indices of the `items` array, it would be just as effective to check if oldIndex/newIndex are greater than or equal to `items.length`

Here's an example:

```
try {
  const oldIndex = parseInt(active.id, 10); 
  const newIndex = parseInt(over.id, 10);

  if (oldIndex >= items.length || newIndex >= items.length)
    return;

  // ... rest of the code ...
} catch (_) { return; }
```

(I usually wrap `parseInt` calls in a try-catch since it has the possibility of throwing but it's not totally necessary)

Reply:
I think I follow, but this might be good to huddle on in case I’m misunderstanding. But, I’m not sure this would work. While the id’s are initially based on each variations index, as we drag and drop, a variations index in the array will change, but it’s id won’t.  For example, if we had an array [“a”, “b”, “c”], `a`’s index and id is 0 and `b`’s index and id is 1.  However, if we swap their positions, `a`’s index is now 1, but it’s id is still 0. So I think we still have to use the `id` in order to get the current index.  I do think we could use ES6 syntax here and instead of using a forLoop use `map` and `indexOf`. Curious if you prefer that 

E.G.

```const oldIndex = draggableVariations
                    .map((variation) => variation.id)
                    .indexOf(active.id);
                  const newIndex = draggableVariations
                    .map((variation) => variation.id)
                    .indexOf(over.id);
  ```